src/main/java/com/palehorsestudios/alone/Shelter.java

```diff
@@ -40,24 +40,20 @@ public int getWaterTank() {
     return waterTank;
   }
 
-  public Result updateWater(int water) {
-    Result.Builder resultBuilder = new Result.Builder();
+  public int updateWater(int water) {
```

Comment:
I think this can be a void method unless the returned value is being used somewhere else I don't see. Just set waterTank based on the logic below. The method could be simplified using a ternary after `waterTank += water`. Something like `waterTank < MIN_WATER ? MIN_WATER : Math.min(waterTank, MAX_WATER);`